Terasology/MetalRenegades
# AI Agents Spawning inside Buildings

## Contains
- Gooey characters spawning inside buildings generated by DynamicCities.
- Each spawnable character now contains an `AgentComponent`.
- Any building that is able to house characters now contains a `PotentialHomeComponent` in the building prefab, which lists the characters currently inside the building, and how many characters a building can provide shelter for.
- When a character spawns inside a building, it is given a `HomeComponent`, which provides a reference to this building.

## How to Test
- This PR depends on Terasology/DynamicCities#43, make sure this change is either merged or included manually or else **the game will crash**.
- Start a game with MetalRenegades enabled.
- When a city spawns, look inside any of the buildings. There should be any number from 1 to 5 gooey characters inside depending on the building.
